pkg/osquery/extension.go

```diff
@@ -517,6 +518,18 @@ func (e *Extension) GenerateConfigs(ctx context.Context) (map[string]string, err
 	} else {
 		// Store good config
 		e.knapsack.ConfigStore().Set([]byte(configKey), []byte(config))
+
+		// open the start up settings writer just to trigger a write of the config,
+		// then we can immediately close it
```

Comment:
I think maybe it would feel cleaner/less easy to introduce bugs in the future if we had a specific function to call to perform the config write, rather than relying on a side effect of `OpenWriter`...maybe `setFlags` could be exposed as `SyncFlags` or something? What do you think?

Reply:
how about renaming to `WriteSettings` and removing from the constructor?

Reply:
That works for me 🙂

Reply:
Done